analyses/reconstruct_transmission/estimate_infections.qmd

```diff
@@ -0,0 +1,270 @@
+---
+title: "How to reconstruct infection dynamics from incidence data on delayed outcomes like deaths"
+format: 
+  html:
+    code-link: true
+editor: source
+editor_options: 
+  chunk_output_type: console
+date: last-modified
+toc: true
+toc_float: true
+---
+
+## Ingredients
+
+- We want to estimate infection dynamics from incidence data on delayed outcomes such as hospitalisations or deaths.
+- What we have:
+  1. Time series of new outcomes (e.g. deaths) per day.
+  2. Estimates of the delay from infection-to-onset and onset-to-death distributions
+- We assume new infections each day have a [prior based on a Gaussian process](https://epiforecasts.io/EpiNow2/articles/estimate_infections.html), which allows for faster estimation from delayed outcome data than modelling the full transmission process (and hence also estimating the time varying reproduction number $R_t$).
+
+
+## Steps in code
+```{r}
+#| warning: false
+
+# Load required packages
+# remotes::install_github("epiforecasts/EpiNow2@dist-interface") # To
+library(EpiNow2)
+
+# Load epiparameter from GitHub
+library(epiparameter)
+
+# Load CRAN packages required for this vignette
+library(dplyr) # required to format outputs
+library(httr) # required to load UK COVID data from URL
+library(cfr) # required for Ebola data (included in this package)
+
+# - - - 
+# Example 1: reconstruct SARS-CoV-2 infection dynamics in the UK from daily data on deaths, 2020
+# - - -
+
+# Define URL for the UK COVID dashboard API
+uk_dashboard_url <- "https://coronavirus.data.gov.uk/api/v1/data?filters=areaType=overview;areaName=United%2520Kingdom&structure=%7B%22areaType%22:%22areaType%22,%22areaName%22:%22areaName%22,%22areaCode%22:%22areaCode%22,%22date%22:%22date%22,%22newDailyNsoDeathsByDeathDate%22:%22newDailyNsoDeathsByDeathDate%22,%22cumDailyNsoDeathsByDeathDate%22:%22cumDailyNsoDeathsByDeathDate%22%7D&format=csv"
+
+# Load data from the UK COVID dashboard API and format as data.frame
+load_uk_data_text <- GET(uk_dashboard_url)
+uk_data <- read.csv(text = content(load_uk_data_text, "text"), stringsAsFactors = FALSE)
```

Comment:
I currently get this output from `uk_data` object. Could you verify if you still get a data frame output?

```
> uk_data
  [1] X..DOCTYPE.html..html.lang.en.class.govuk.template.__..(...)   
  [2] queries...state..data..success.true    
  [3] data...status.Green    
  [4] geography_name.North.East    
  [5] geography_code.E12000001    
  [6] refresh_date.null.   
  [7] X.status.Green    
  [8] geography_name.North.West    
  [9] geography_code.E12000002    
 [10] refresh_date.null..1    
 [11] X.status.Green.1    
 [12] geography_name.Yorkshire.and.The.Humber    
 [13] geography_code.E12000003    
 [14] refresh_date.null..2    
 [15] X.status.Green.2    
 [16] geography_name.East.Midlands    
 [17] geography_code.E12000004    
 [18] refresh_date.null..3    
 [19] X.status.Green.3    
 [20] geography_name.West.Midlands    
 [21] geography_code.E12000005    
 [22] refresh_date.null..4    
 [23] X.status.Green.4    
 [24] geography_name.East.of.England    
 [25] geography_code.E12000006    
 [26] refresh_date.null..5    
 [27] X.status.Green.5    
 [28] geography_name.London    
 [29] geography_code.E12000007    
 [30] refresh_date.null..6    
 [31] X.status.Green.6    
 [32] geography_name.South.East    
 [33] geography_code.E12000008    
 [34] refresh_date.null..7    
 [35] X.status.Green.7    
 [36] geography_name.South.West    
 [37] geography_code.E12000009    
 [38] refresh_date.null... 
 [39] dataUpdateCount.1    
 [40] dataUpdatedAt.1717579083479    
 [41] error.null    
 [42] errorUpdateCount.0    
 [43] errorUpdatedAt.0    
 [44] fetchFailureCount.0    
 [45] fetchFailureReason.null    
 [46] fetchMeta.null    
 [47] isInvalidated.false    
 [48] status.success    
 [49] fetchStatus.idle.   
 [50] queryKey..weather.alert.list    
 [51] heat.   
 [52] queryHash...weather.alert.list...heat..
```

Reply:
elegant! thanks for sharing the new features